sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_appinsights_utils.py

```diff
@@ -0,0 +1,535 @@
+    default_workspace = client.resources.list_by_resource_group(
+        default_resource_group,
+        filter="substringof('%s',name)"%get_default_log_analytics_name(subscription_id, location)
+    )
+    for item in default_workspace:
+        # return arm id of log analytics workspace, true for is_existing
+        return (item.id, True)
+    # else return arm id for to be created log analytics workspace, false for is_existing
+    return (
+        '/subscriptions/%s/resourceGroups/%s/providers/Microsoft.OperationalInsights/workspaces/%s'%(subscription_id, default_resource_group, get_default_log_analytics_name(subscription_id, location)), 
+        False)
+
+
+def get_log_analytics_deployment(deployment_name: str, location: str, subscription_id: str) -> dict:
+    return {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2019-10-01",
+            "name": deployment_name,
+            "resourceGroup": get_default_resource_group_name(location),
+            "subscriptionId": subscription_id,
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "parameters": {},
+                    "variables": {},
+                    "resources": [
+                        {
+                            "apiVersion": "2020-08-01",
+                            "name": get_default_log_analytics_name(subscription_id, location),
+                            "type": "Microsoft.OperationalInsights/workspaces",
+                            "location": "[if(or(equals(parameters('applicationInsightsLocation'),'westcentralus'), equals(parameters('applicationInsightsLocation'),'eastus2euap'), equals(parameters('applicationInsightsLocation'),'centraluseuap')),'southcentralus',parameters('applicationInsightsLocation'))]",
+                            "properties": {}
+                        }
+                    ]
+                }
+            }
+        }
+
+
+def get_default_log_analytics_name(subscription_id: str, location: str) -> str:
+    logWorkspaceName = "DefaultWorkspace-%s-%s"%(subscription_id,get_la_region_code(location))
+    return logWorkspaceName if len(logWorkspaceName) < 63 else logWorkspaceName[0,62]
+
+
+def get_default_resource_group_name(location: str) -> str:
+    return "DefaultResourceGroup-%s"%get_la_region_code(location)
+
+
+def get_la_region_code(location: str) -> str:
+    if location in ["westcentralus", "eastus2euap", "centraluseuap"]:
+        return region_mapping["regions"]["southcentralus"]["laRegionCode"]
+    try: 
+        region = region_mapping["regions"][location.lower()]
+        return region["laRegionCode"] if region["laRegionCode"]!="" else location
+    except: 
+        return location
+
+
+region_mapping = {
```

Comment:
It's scary to see this in the SDK code. Will this mapping ever change? Also does the UX have the same mapping so that the workplace creation experience is consistent between the CLI/SDK and the UX? I feel it may be worth to create a Rest API for this.

Reply:
Yes, I suppose that brings a good point. I was following the practice of the Portal UI and Studio where these values are manually included within the code. This mapping should not change for existing regions, but additions would happen for new regions. [Portal](https://dev.azure.com/msdata/Vienna/_git/vienna?path=/IbizaExtension.MLServices/MLTeamAccounts/MLTeamAccounts/Client/Resource/Create/Utilities/AppInsightsUtilities.ts&version=GBmaster&line=86&lineEnd=86&lineStartColumn=1&lineEndColumn=45&lineStyle=plain&_a=contents) and [ML Studio](https://dev.azure.com/msdata/Vienna/_git/workspace-portal?path=/common/core/src/context/workspace/AppInsights.utils.ts&version=GCac30c6c3c3ba1494ecfaa8dcbc21900ab8c2eaa5&line=55&lineEnd=55&lineStartColumn=1&lineEndColumn=45&lineStyle=plain&_a=contents). These values are used for the naming of resources. I think this is not within scope of creating a REST API to fetch this information since these values are not defined by AzureML. I guess an alternative approach would be to change the naming convention for these resources to something like "DefaultWorkspace-{subscriptionId}-{location}" instead of "DefaultWorkspace-{subscriptionId}-{laRegionCode}" but this is not the approach taken by Portal/Studio. I could try to see if updates on that end would be possible, what are your thoughts?

Reply:
A concern is that we should continue to use the same naming convention used for previously created resources since the goal in this is to reuse the same Log Analytics resource across many AzureML deployments. If we change the naming convention, that would not be the case. I see both sides here but am not sure how to proceed.

Reply:
Actually, this naming convention will not work "DefaultWorkspace-{subscriptionId}-{location}" due to the limitation that each Log Analytics Workspace name must be between 4 and 63 characters in length. With this, the current DefaultWorkspace-{subscriptionId}- uses 54 of those characters. Thus, locations over 9 characters will violate that rule, causing failure in creating resource (ie. northcentralus)

Reply:
Updated naming convention to "DefaultWorkspace-{location}" and removed region mapping, simplifying names for resources and removing external logic/dependencies